core/src/main/scala/org/apache/spark/shuffle/ShuffleMemoryManager.scala

```diff
@@ -175,7 +175,9 @@ private[spark] object ShuffleMemoryManager {
     val minPageSize = 1L * 1024 * 1024   // 1MB
     val maxPageSize = 64L * minPageSize  // 64MB
     val cores = if (numCores > 0) numCores else Runtime.getRuntime.availableProcessors()
-    val safetyFactor = 8
+    // Because of rounding to next power of 2, we may have safetyFactor as 8 in worst case
```

Comment:
This is a new comment. For safetyFactor = 16, then size could be doubled by rounding, so it seems like having a safetyFactor as 8.